File: python/euler/bin/p0016.py
# project euler: problem 16

# number of digits of 2^1000
#   1 + floor(log10(2**1000))
#     = 1 + floor(1000 * log10(2))
#     = 1 + floor(1000 * 0.30102...)
#     = 1 + 301
#     = 302


def compute(num: int) -> str:
    return str(sum(map(int, str(pow(2, num)))))


def solve() -> str:
    return compute(1000)

File: clojure/src/project_euler/sol/p0016.clj
(ns project-euler.sol.p0016
  (:require
   [project-euler.lib.math :as my-math]
   [project-euler.lib.util :as util]))

(defn solve
  ([]
   (solve 1000))
  ([exp]
   (apply + (util/digits (my-math/pow 2 exp)))))
